types/currency_test.go

```diff
@@ -298,3 +299,30 @@ func TestNegativeNewCurrency(t *testing.T) {
 	negBig := big.NewInt(-1)
 	_ = NewCurrency(negBig)
 }
+
+// TestCurrencyUint64 tests that a currency is correctly converted to a uint64.
+func TestCurrencyUint64(t *testing.T) {
+	// Try a set of valid values.
+	attempts := []uint64{0, 1, 2, 3, 4, 25e3, math.MaxUint64}
```

Comment:
not a great name. I usually just use `tests`, but you could also use `u64s`, `currency64s`, `values`, etc.